packages/eslint-plugin/docs/rules/require-await.mdx

```diff
@@ -21,5 +25,11 @@ async function returnsPromise1() {
   return Promise.resolve(1);
 }
 
+async function* asyncGenerator() {
```

Comment:
Ah - so, the trouble now, is, this added example is correct in the base rule, too. So, we'd want to add an additional _incorrect_ example (in a separate tab/section - see other rules for examples) that _will_ flag.

Reply:
When I added the example I checked that the base rule doesn't support that.
[ESLint playground(https://eslint.org/play/#eyJ0ZXh0IjoiYXN5bmMgZnVuY3Rpb24gc29tZXRoaW5nKCkge1xuICBjb25zdCBudW0gPSBQcm9taXNlLnJlc29sdmUoMSk7XG4gIHJldHVybiAxICsgbnVtO1xufVxuXG5hc3luYyBmdW5jdGlvbiogYXN5bmNHZW5lcmF0b3IoKSB7XG4gIGNvbnN0IG51bSA9IFByb21pc2UucmVzb2x2ZSgxKTtcbiAgeWllbGQgMSArIG51bTtcbn1cbiIsIm9wdGlvbnMiOnsicnVsZXMiOnsicmVxdWlyZS1hd2FpdCI6WyJlcnJvciJdfSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=)
[Typescript-eslint playground](https://typescript-eslint.io/play/#ts=5.5.2&fileType=.ts&code=IYZwngdgxgBAZgV2gFwJYHsIxOgtgU2QAtUIBzACgEoYBvAWACgYYpMRkYIFcYBeGAAUATnlQh8AOmH4cAGwBu%2BCgEYqAbiYsZyBMKwqYAai49NjAL5MmoSLEQoMEAFQxb0AOL4I%2BYcGTowtR0WqzsnNy8AiJiEtKy6IrKauYsYKj4cgAmMIYmkeZWjExAA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEI0QR1gEtpEBaAQwHdzCAXdKAewDNnIAacbSAAVoE8ADigDG0QoNqkU8QgDtaAehIFiZKjXoZIiaNEbRIXAL4hjQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

Reply:
I'm not following, sorry. Could you clarify?

Reply:
I clarified this in the comment above. Could you check the links and see the difference?

Reply:
Oh, looking at this now, I think I looked only in the "correct" tab, here, and failed to see the corresponding code in the "incorrect" tab. In any case this looks good. Sorry for the confusion!